portfolio/src/main/webapp/script.js

```diff
@@ -133,7 +133,10 @@ function getComments() {
   const maxComments = document.getElementById('max-comments').value;
   if (maxComments === null) maxComments = 5; /* Show at most 5 comments by default. */
 
-  fetch(`/comments?max-comments=${maxComments}`).then(response => response.json()).then((comments) => {
+  fetch(`/comments?max-comments=${maxComments}`)
+  .then(response => response.json())
+  .then((response) => {
+    const comments = response.comments;
```

Comment:
you should check the value of response.success and add comments only if it is true. If it is false then you want to display the error message.